tox.ini

```diff
@@ -27,7 +28,7 @@ commands =
     # https://docs.openstack.org/bandit/latest/blacklists/blacklist_imports.html#b404-import-subprocess
     # https://docs.openstack.org/bandit/latest/plugins/subprocess_popen_with_shell_equals_true.html
     # https://docs.openstack.org/bandit/latest/plugins/subprocess_without_shell_equals_true.html
-    bandit -r in_toto --skip B404,B603
+    bandit -r in_toto --skip B107,B404,B603
```

Comment:
I had to add skipping of B107 because otherwise bandit would complain that I'm hardcoding a password wherever I have used `""` as the default parameter for password (in order to prevent prompts for password). Any workarounds around this? Using `None` as the default parameter doesn't throw errors but this does? :thinking:

Reply:
A shortcoming of running Bandit as above (`bandit -r in_toto --skip B107,B404,...`) is that --skip is used for every single unit test.  Assuming that you only need to exclude B107 for certain tests, it might be better to create a Bandit profile for unit tests that require custom settings.  

You can take a look at the Bandit docs to see how profiles can be created in a configuration file.
https://bandit.readthedocs.io/en/latest/config.html?highlight=profile

Here is an example configuration file for Bandit that uses profiles:
https://github.com/chair6/bandit/blob/4004374a894bb41414ce07546518ac131353ec36/bandit.yaml#L21-L26

Reply:
Bandit is a linter? It doesn't run on specific tests. What we are doing right now is recursively running it on all files in the in_toto folder. The rule I'm skipping is this one which itself claims to throw a lot of false positives:
https://docs.openstack.org/bandit/latest/plugins/b107_hardcoded_password_funcdef.html

I'm adding a note regarding that.

Reply:
Bandit is a security linter, if I'm not mistaken.  It says so on their project page: "A security linter from OpenStack Security"

https://pypi.org/project/bandit/

I was recommending that you run Bandit + a profile on specific tests that need to avoid B107, or whatever.  For instance,

```
$ bandit tests/test_in_toto_verify.py -p ignore_pw_test
(other bandit invocations go here, with perhaps different profiles)
```

Reply:
Note: You don't have to use profiles if you wish, but I only recommend it so that we don't ignore the B107 test across the entire codebase.  I'm not sure if there is a better way to do it, but using profiles is one possible way to deal with an overreaching --skip.